### DISCUSSION TITLE
Can anyone lend some wisdom to setting up custom icons on the V2?

### DISCUSSION THREAD

**Discussion by hayhay7789** [Jyers/Marlin#409] [Q&A] [answered]
I am a recent buyer of the Creality Ender 3 v2, purchased from Amazon. I was referred to this firmware by a Redditor who said the developers may be able to provide information to resolve my issue since custom icons work on this fork. 

I was able to update the mainboard to the latest prebuilt firmware available on here without any problems. I then edited the icons in the "original" folder so they would be purple instead of blue by changing the image hues, compiled them into an ICO file for the printer, then loaded them onto an SD card in the DWIN_SET folder to update the display with the new file. The only problem is, I have been having trouble with getting the icons to show up correctly. Some of the icons, like 003_ICON_Prepare_0.jpg, work and look great, but others, like 021_ICON_Continue_0.jpg, do not display, and are blank spaces on my Ender display. At first, none of the icons displayed, but I learned that the jpeg files must be baseline, not progressive, so I saved them all in Microsoft Paint to correct this. This partially fixed the problem, as some of the icons started displaying, but many still do not. Only 20 or so of them are working so far out of the ~90 icons. I was hoping someone maybe has knowledge about either the Ender 3v2 or about jpeg files so that I may fix whatever is causing them to not work correctly. Perhaps I need to change the colour space because the printer cannot display the colours in the non-working images? I double-checked that the images are all now baseline JPEGs so I'm pretty stumped. If someone with technical knowledge could share some wisdom, I would be very grateful.

I have attached some [pictures](https://imgur.com/a/uXrhnik) of my Ender display with the icons not displaying correctly, and the two icon files I mentioned above (one working, one not working). I took these pictures when I was still running Marlin, not this fork, but the issue persists on this firmware. Maybe there's some differing data on the files which will offer solutions? (EXIF data, as I expected, didn't help)

Thank you for this great fork

**Comment by Jyers** [comment, marked as answer]
I'm not an expert as most of the custom icon work has come from PR's
If they're around @corben2 might be able to shed some light on the issue.

**Reply by hayhay7789** [reply]
Thanks, hopefully he can provide some insight! :)

**Reply by corben2** [reply]
I don't know a whole lot about jpegs or image formats in general, I looked at the example images your provided and don't see anything obvious for why they wouldn't work. 

I adjusted the hue of mine, using Imagemagick, then loaded them onto my LCD and they all seem to work. Here's the Imagemagick command I used: `convert $file -modulate 100,100,130 $file` where $file is the image. I looped over all my images with that same command.

I pushed these images, with the new 9.ICO, up to a new branch [here](https://github.com/corben2/Marlin/tree/IconColorTest/Display%20firmware/dwin-ico-tools). The changed images are under new+denoised and the new ICO file is called 9.new+denoised.ICO. Also under that branch, is the [DWIN_SET folder (with the changed 9.ICO) that you load on the SD to flash,](https://github.com/corben2/Marlin/tree/IconColorTest/Display%20firmware/DWIN_SET) if you want to try it. Maybe those can be of some help, I'm not sure.

Maybe someone else that has more knowledge of JPEGs can shed some light on this.

**Comment by Jonny11bonk** [comment]
I still have the problem. Im using paint.net, I have created the post in the ISSUES.
